streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -151,7 +153,14 @@ public KStreamBuilder() {
         addProcessor(name, processorSupplier, source);
 
         final KTableImpl kTable = new KTableImpl<>(this, name, processorSupplier, Collections.singleton(source), keySerde, valSerde, storeName);
-        kTable.materialize((KTableSource) processorSupplier);
+        StateStoreSupplier storeSupplier = new RocksDBKeyValueStoreSupplier<>(storeName,
```

Comment:
We should also update the JavaDoc for this parameter, currently it is `the state store name used if this KTable is materialized, can be null if materialization not expected`.